src/specifications/profile-resolution/profile-resolution-specml.xml

```diff
@@ -3,1772 +3,1370 @@
-  &lt;matching pattern="ac*"/>&lt;!-- matches any control whose ID matches glob expression 'ac*' -->
-&lt;/include-controls>
-</tagging>
-            <p><revisit>The match pattern is evaluated as a regular expression using XPath regular
-                expression syntax. [XXXX]</revisit></p>
-          </section>
-          <section id="include-all">
-            <head>Selecting all controls</head>
-            <p>Select all controls from a catalog by using an <src>include-all</src> directive:</p>
+        <head>Including Controls</head>
+        <p>Each import contains directives on which parts of the imported Catalog are to be
+          included. Throughout the rest of this document, we will refer to "explicit" vs "implicit"
+          inclusion. Explicit Inclusion refers to anything that has been referenced directly by any
+          of the below directives. Implicit Inclusion refers to objects that have been included due
+          to other circumstances, but that have not been referenced by the inclusion directives. </p>
 
-            <tagging whose="source_profile">&lt;include-all/>
-</tagging>
-          </section>
-          <section id="include-child-controls">
-            <head>Including child  and descendantcontrols</head>
-            <p>In OSCAL, controls may contain controls. For example, in SP 800-53 many controls are
-              supplemented with control enhancements; in OSCAL these are represented as child
-              controls within parent controls. So parent AC-2 (in a given catalog) has children
-              AC-2(1) through AC-2(13), for example.</p>
-            <p>Child controls can be included by the same mechanism as controls, i.e. by means of an
-              ID call. Alternatively a match can frequently be applied (at least given most ID
-              assignment schemes) to match controls and child controls together.</p>
-            <p>Additionally, a <src>@with-child-controls</src> directive on an
-                <src>include-controls</src> indicate that <em>all descendant controls </em>should be
-              included with the applicable call(s) or match(es). Its operation is recursive in this
-              respect: <src>@with-child-controls='no'</src> applies not only to controls included
-              specifically, but also to those selected by means of the flag applying to their parent
-              (containing) controls.</p>
-            <p>Furthermore, <src>include-all[@with-child-controls='no']</src> may select all
-              controls placed directly within a group, excluding all controls appearing inside other
-              controls.</p>
+        <p>It is RECOMMENDED that implementations track which objects are explicitly included vs
+          implicitly included as the intermediate is built out. This information is required to be
+          used during the pruning process <xref rid="cleanup"/>. </p>
+        <section id="include-all">
```

Comment:
If it is optional to track explicit vs implicit inclusion, it cannot be required to used this information during the pruning process. Perhaps we are saying that a tool MUST be able to track which objects are included implicitly vs explicitly for use in pruning.